source/javascripts/components/unified-editor/WorkflowConfig/hooks/useRenameWorkflow.ts

```diff
@@ -3,38 +3,52 @@ import useBitriseYmlStore from '@/hooks/useBitriseYmlStore';
 import { useWorkflowConfigContext } from '../WorkflowConfig.context';
 
 const useRenameWorkflow = (onChange?: (newWorkflowId: string) => void) => {
-  const { id: currentWorkflowId } = useWorkflowConfigContext() ?? { id: '' };
+  const selectedWorkflowId = useWorkflowConfigContext()?.id ?? '';
+  const workflowIdsInTheStore = useBitriseYmlStore((s) => Object.keys(s.yml.workflows ?? {}));
 
   const [isRenaming, setIsRenaming] = useState(false);
-  const [previousWorkflowId, setPreviousWorkflowId] = useState(currentWorkflowId);
+  const [nextWorkflowId, setNextWorkflowId] = useState(selectedWorkflowId);
+  const [prevWorkflowId, setPrevWorkflowId] = useState(selectedWorkflowId);
 
   const { createWorkflow, renameWorkflow, deleteWorkflow } = useBitriseYmlStore((s) => ({
     createWorkflow: s.createWorkflow,
     renameWorkflow: s.renameWorkflow,
     deleteWorkflow: s.deleteWorkflow,
   }));
 
-  const renameCallback = useCallback(
-    (newWorkflowId: string) => {
-      setIsRenaming(true);
-      renameWorkflow(currentWorkflowId, newWorkflowId);
-      createWorkflow(currentWorkflowId, newWorkflowId);
-      setPreviousWorkflowId(currentWorkflowId);
+  const isNewWorkflowPersisted = workflowIdsInTheStore.includes(nextWorkflowId);
+  const isNewWorkflowSelected = prevWorkflowId !== nextWorkflowId && nextWorkflowId === selectedWorkflowId;
```

Comment:
Why do we need this `prevWorkflowId !== nextWorkflowId`?

Reply:
We don't need this. Removed from `useRenameWorkflow` and `useRenamePipeline`, thanks!